lib/rules/redux-no-bound-selectors.js

```diff
@@ -0,0 +1,65 @@
+/**
+ * @fileoverview Disallow new functions in react-redux#connect
+ * @author Automattic
+ * @copyright 2017 Automattic. All rights reserved.
+ * See LICENSE.md file in root directory for full license.
+ */
+'use strict';
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+const REF = ' See wp-calypso#14024';
+const BIND_ERROR_MESSAGE = "Don't bind functions within `connect`." + REF;
+const FUNC_ERROR_MESSAGE = "Don't instantiate functions within `connect`." + REF;
+
+function isStateInArgs( node ) {
+	return node.params.some( function( paramNode ) {
+		return paramNode.name === 'state';
```

Comment:
Matching by assigned name could have some false positives, but the alternative would be pretty cumbersome I think (tracking assigned `connect` named import to its first function argument when invoked). Wondering if people would use [React's `this.setState`](https://facebook.github.io/react/docs/react-component.html#setstate) function callback form with `state` as the argument name.